lib/livebook_cli/task.ex

```diff
@@ -1,11 +1,78 @@
 defmodule LivebookCLI.Task do
+  import LivebookCLI.Utils
+
   @doc """
   Returns a description of the task usage.
   """
-  @callback usage() :: String.t()
+  @callback usage() :: IO.chardata()
 
   @doc """
   Runs the task with the given list of command line arguments.
   """
   @callback call(args :: list(String.t())) :: :ok
+
+  @doc """
+  Runs the task with the given list of command line arguments.
+  """
+  @spec call(String.t(), list(String.t())) :: :ok
+  def call(name, args) do
+    task = fetch_task!(name)
+    task.call(args)
+  rescue
+    exception -> log_exception(exception, name, __STACKTRACE__)
+  end
+
+  @doc """
+  Shows the description of the task usage.
+  """
+  @spec usage(String.t()) :: :ok
+  def usage(name) do
+    task = fetch_task!(name)
+    print_text(task.usage())
+  rescue
+    exception -> log_exception(exception, name, __STACKTRACE__)
+  end
+
+  defp fetch_task!("server"), do: LivebookCLI.Server
+  defp fetch_task!("deploy"), do: LivebookCLI.Deploy
+
+  defp fetch_task!(name) do
+    log_error("Unknown command #{name}")
+    print_text(LivebookCLI.usage())
+
+    System.halt(1)
+  end
+
+  defp log_error(message) do
+    [:red, message]
+    |> IO.ANSI.format()
+    |> IO.puts()
+  end
+
+  @spec log_exception(Exception.t(), String.t(), Exception.stacktrace()) :: no_return()
+  defp log_exception(exception, command_name, stacktrace) when is_exception(exception) do
+    [:red, format_exception(exception, command_name, stacktrace)]
+    |> IO.ANSI.format()
+    |> IO.puts()
+
+    System.halt(1)
+  end
+
+  defp format_exception(%OptionParser.ParseError{} = exception, command_name, _) do
+    """
+    #{Exception.message(exception)}
+
+    For more information try:
+
+       livebook #{command_name} --help
+    """
+  end
+
+  defp format_exception(%RuntimeError{} = exception, _, _) do
```

Comment:
Unfortunately regular Elixir code can also emit RuntimeError, and it would get mixed up with this. If you want this functionality, I would suggest defning a `LivebookCLI.Error` and raise that instead.